### PR TITLE
Unit testing data.js

### PR DESCRIPTION
**Pull request by stephenisau**
This pull request includes a

- [ ] Bug fix
- [X] New feature
- [X] Testing
- [ ] Translation

The following changes were made

- Added tests folder with unit sub-folder
- Added mocks folder to mock methods built-in to the fs module (specifically fs.readdirSync, fs.readFileSync to return custom data)
- Included path import in 'models/data.js'
- Exported functions from data.js
- Added unit tests to `__tests__/unit/data.test.js`
- Completed TODO related to adding try-catch block in reader_norm function L54-L102
- Completed TODO related to making pathnames relative to the file
- Completed TODO related to adding utf8 encoding argument on L34 and L114

**Comments:**
- I wanted to use a __mocks__ directory to manually mock methods built-in to node's filesystem module. Two of the functions relied heavily on reading text files from directories and processing data from these directories. Manually mocking the methods in charge of reading the filesystem and the corresponding files allowed for me to have better control over the input/output, and provide more flexibility with tests. I used this as an example: (https://jestjs.io/docs/en/manual-mocks.html)

If this is related to an existing ticket, include a link to it as well.
Issue [#5](https://github.com/rithmschool/numbers_api/issues/5)

### CODE REVIEW

File: models/data.js [hunk 1]
```diff
@@ -253,34 +238,38 @@ reader_norm(exports.math, "models/math/norm/", function (element) {
...
-        console.log("Missing: " + config.category + ": " + index);
+        console.warn(`Missing: ${config.category} : ${index}`);
       }
     });
   });
 })();
+
+exports.reader_norm = reader_norm;
+exports.reader_manual = reader_manual;
+exports.normalize_common = normalize_common;
```
Comment (mmmaaatttttt) [inline models/data.js:275]: can you change this to:

```suggestion
module.exports = {
  reader_norm,
  reader_manual,
  normalize_common
}
```

if there's anything else exported in the file, let's move the exporting to the bottom as well.

**Review by mmmaaatttttt — CHANGES_REQUESTED** [review]
just a small change here @stephenisau - thanks!

fyi in general it's better to leave PRs open after feedback, instead of closing and opening a new one, since keeping the same PR open preserves the history of the feedback.

File: models/data.js (hunk 1 above)
Comment (stephenisau) [reply models/data.js:275]: Okay, I'll keep that in mind for future reference. Thanks for reviewing this @mmmaaatttttt! I've made the above changes and pushed it to my version of the branch. 

Can we merge this PR? I'll try look for the differences with PR #34 and #35.

Comment (mmmaaatttttt) [reply models/data.js:275]: Hey @stephenisau, I don't see any new commits on this branch. Did you push up to the correct branch? The PR will auto-update with any new commits when you push them up. Once the commit is on this branch i'll merge :)

Comment (stephenisau) [reply models/data.js]: @mmmaaatttttt just saw this but I think my commit was stuck on my local branch. I just pushed this up to the git branch on my repo!

**Review by mmmaaatttttt — APPROVED** [review]

**Pull request merged**